### PR TITLE
Intel OneAPI

### LINKED ISSUE #43: Intel OneAPI Containers
Look into the possibility of building and distributing Intel OneAPI development containers.  This may be a way to provide a free intel development environment, including compilers and intel MPI, as an alternative to Parallel Studio.

### PR DESCRIPTION
**Pull request by mmiesch**
## Description

This PR accompanies https://github.com/JCSDA-internal/jedi-tools/issues/134 - see description there for further details and for instructions on how to access the new containers.

This eliminates the old intel 17 container and adds a new intel-OneAPI container.  It also splits the build and push of a container into two separate steps / scripts for ease of testing.  And it adds a `tag_release` script for tagging release versions of containers and saving them for posterity / reproducibility.

### Issue(s) addressed

Link the issues to be closed with this PR
- fixes #43

### CODE REVIEW

File: build_public_container.sh [hunk 1]
```diff
@@ -41,68 +41,39 @@ echo "=============================================================="
 mkdir -p context
 cd context
 ln -sf ../Dockerfile.${CNAME} .
-$SUDO docker image build --no-cache --pull -t ch-${CNAME} -f Dockerfile.${CNAME} .
+#$SUDO docker image build --no-cache --pull -t ch-${CNAME}:${TAG} -f Dockerfile.${CNAME} .
```
Comment (rickgrubin) [inline build_public_container.sh]: Is this supposed to be commented out?

**Review by rickgrubin — APPROVED** [review]
Really nice; LGTM

File: build_public_container.sh (hunk 1 above)
Comment (mmiesch) [reply build_public_container.sh]: good catch!

**Review by markjolah — APPROVED** [review]

**Comment by mmiesch** [comment]
I know this is already approved but I'm going to close it and re-open it later in light of [this issue](https://github.com/JCSDA-internal/containers/issues/49)

**Pull request closed without merging**